app/models/user.server.ts

```diff
@@ -8,6 +8,13 @@ export interface User extends PrismaUser {
   // the base domain to use for all DNS records created by this user
   // (e.g., jsmith.starchart.com with records using *.jsmith.starchart.com)
   baseDomain: string;
+  /* the effectiveUsername, this is for admin impersonating users. Regular
+   users will have an effectiveUsername that is always the same as their
+   username, admins (identified by group) will start with an empty string
+   that can be changed to another users username on the admin portal for 
+   them to impersonate */
+  effectiveUsername?: User['username'];
```

Comment:
I don't think we do need it since everything kind of comes from the cookie now, is there a better way of doing this than with the cookie? Maybe keeping track of it on this property we are thinking about removing, but this is dynamically appended every time we load same as baseDomain so that does not make sense. Anyways, I am ok with removing it I think.

Reply:
Honestly it's kind of hard keeping track of stuff